content/docs/use-cases/versioning-data-and-model-files/index.md

```diff
@@ -1,77 +1,121 @@
+are designed for source code management (SCM) however, and thus ill-equipped to
+support data science needs. That's where DVC comes in: with its built-in data
+<abbr>cache</abbr>, reproducible [pipelines](/doc/start/data-pipelines), among
+several other novel features (see [Get Started](/doc/start/) for a primer.)
+
+## Track data and models for versioning
+
+Let's say you already have a <abbr>DVC repository</abbr> and put a bunch of
+images in the `images/` directory. Then you build a `model.pkl` based on them.
 
 ```dvc
-$ ls images
+$ ls images/
 0001.jpg 0002.jpg 0003.jpg 0004.jpg ...
-
 $ ls
-model.pkl
+images model.pkl
 ```
 
-To start using DVC we need to [initialize](/doc/command-reference/init) a
-<abbr>DVC project</abbr> on top of the existing Git repo:
+Start tracking the dataset and the model file with `dvc add`:
 
 ```dvc
-$ dvc init
+$ dvc add images/
+$ dvc add model.pkl
 ```
 
-Start tracking the images directory and the model with `dvc add`:
+This generates `.dvc` files, and puts the originals in `.gitignore`. Commit this
+project's version:
 
 ```dvc
-$ dvc add images
-$ dvc add model.pkl
+$ git add images.dvc model.pkl.dvc .gitignore
+$ git commit -m "Track images and model with DVC."
+```
+
+## Track pipeline artifacts for versioning
+
+Some of DVC's most important features allow for defining the processes to build
+artifacts such as ML models in a simple `dvc.yaml` file, in order to run and
+reproduce them later.
+
+> See [Data Pipelines](/doc/start/data-pipelines) for more information.
+
+Instead of training the model file on your own and adding the `model.pkl` to DVC
+manually, we can add only the images dataset in the previous step, and use this
+`dvc.yaml`:
+
+```yaml
+stages:
+  train:
+    cmd: python train.py images/
+    deps:
+      - images # Already tracked by DVC
+    outs:
+      - model.pkl
 ```
 
-> Refer also to `dvc run` for more advanced ways to version data and data
-> processes.
+> The file can be written manually or generated with `dvc run`.
 
-Commit your changes:
+`dvc repro` can now execute the above stage for you. DVC will track all of its
+outputs (`outs`) automatically, which get listed in `.gitignore`. Let's do that,
+and commit this project version:
 
 ```dvc
-$ git status
+$ dvc repro
+Running stage 'train' with command:
+        python train.py images/
+Updating lock file 'dvc.lock'
 ...
-Untracked files:
-    .gitignore
-    images.dvc
-    model.pkl.dvc
 
-$ git add images.dvc model.pkl.dvc .gitignore
-$ git commit -m "Track images and model with DVC"
+$ git add dvc.yaml dvc.lock .gitignore
+$ git commit -m "Train model via DVC."
+$ git tag -a "v1.0" -m "Fist model via DVC" # We'll use this soon ;)
 ```
 
-There are two ways to get to the previous version of the dataset or model: a
-full <abbr>workspace</abbr> checkout, or checkout of a specific data or model
-file. Let's consider the full checkout first. It's quite straightforward:
+> See also `dvc.lock`.
+
+## Switching versions
```

Comment:
we don't need 3-4 technical sections, we need to give some sense how does it look like, but it's not "how to".

Reply:
OK, in order to remove more stuff I need to make sure whatever I remove here is covered by the tutorial so I'll review both now ⏳

Reply:
OK, it wasn't that difficult. See latest version please. I think this is pretty much the shortest it can be without completely removing the 2 technical sections left now, I think.